### PR TITLE
122 lookup objects retry

### PR DESCRIPTION
**Pull request by if0s**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.6.0 (November 04, 2022)
+* Added configurable retries on file upload to platform in `Download Files` action
+* Update sailor version to 2.7.1
```
Comment (A3a3e1) [inline CHANGELOG.md:3]:
```suggestion
* Update Sailor version to 2.7.1
```

File: README.md [hunk 2]
```diff
@@ -132,6 +132,9 @@ Finds a file by criteria in the provided directory and uploads (streams) to the
     * **Emit Individually** - Each object will be emitted separately filling the entire message
 * **Number of search terms** - (number, optional, between 0 and 99): Defines the number of search terms that the entity must match
 * **Upload files to attachment** - (dropdown, optional): If set to `Yes` files will be uploaded to platform storage
+* **File Upload Retry** - (number, optional, default 5): How many times to retry file upload as attachment to platform storage
+* **Retry Timeout** - (number, optional, default 10000): How long to wait between retry attempts in milliseconds
+* **File Upload Timeout** - (number, optional, default 10000): If file upload process will be longer than specified number of milliseconds and will not processing any data (receiving or uploading), timeout will be thrown (process will be retried if \"File Upload Retry\" set)
```
Comment (A3a3e1) [inline README.md:137]:
```suggestion
* **File Upload Timeout** - (number, optional, default 10000): If a file upload process is longer than the specified number of milliseconds and is not processing any data (receiving or uploading), the timeout will be thrown (the process will be retried if \"File Upload Retry\" set)
```

File: spec/actions/lookupFiles.spec.js [hunk 3]
```diff
@@ -42,7 +43,7 @@ describe('Lookup Files', () => {
     getReadStreamStub = await sinon.stub(Sftp.prototype, 'getReadStream');
     existsStub = await sinon.stub(Sftp.prototype, 'exists');
     uploadAttachmentStub = await sinon.stub(AttachmentProcessor.prototype, 'uploadAttachment');
-    await lookupFiles.init(cfg);
+    // await lookupFiles.init(cfg);
```
Comment (shulkaolka) [inline spec/actions/lookupFiles.spec.js:46]: delete comments, please

File: spec/actions/lookupFiles.spec.js [hunk 4]
```diff
@@ -97,7 +98,7 @@ describe('Lookup Files', () => {
   });
 
   after(async () => {
-    await lookupFiles.shutdown(cfg);
+    // await lookupFiles.shutdown(cfg);
```
Comment (shulkaolka) [inline spec/actions/lookupFiles.spec.js:101]: the same

**Review by shulkaolka — APPROVED** [review]

File: lib/actions/lookupObjects.js [hunk 5]
```diff
@@ -80,11 +90,34 @@ class LookupFiles extends LookupObjects {
...
+          fileUploadResult = await Promise.race([
+            uploadFromSftpToAttachmentBuffer(this, file, msg.body[DIR], token),
+            timeOut(fileUploadTimeout, token),
+          ]);
+          break;
+        } catch (e) {
+          currentTry++;
+          if (currentTry > fileUploadRetry) throw e;
+          this.logger.error(`got error ${e.message}, going to retry (${currentTry} of ${fileUploadRetry}) upload file`);
+          if (sftpClient) sftpClient.end();
```
Comment (A3a3e1) [inline lib/actions/lookupObjects.js:109]: await?

File: spec/actions/lookupFiles.spec.js (hunk 3 above)
Comment (if0s) [reply spec/actions/lookupFiles.spec.js]: done

File: spec/actions/lookupFiles.spec.js (hunk 4 above)
Comment (if0s) [reply spec/actions/lookupFiles.spec.js]: done

**Review by A3a3e1 — APPROVED** [review]

**Pull request merged**